web/components/search/query-contracts.ts

```diff
@@ -0,0 +1,15 @@
+import { Contract } from 'common/contract'
+import { useEffect, useState } from 'react'
+import { searchClient, searchIndexName } from 'web/lib/service/algolia'
+
+const index = searchClient.initIndex(searchIndexName)
+
+export function useMarketSearchResults(query: string) {
+  const [results, setResults] = useState([] as Contract[])
+  useEffect(() => {
+    index
+      .search(query, { hitsPerPage: 20 })
+      .then((data) => setResults(data.hits as any))
+  }, [query])
```

Comment:
This should have e.g. the same nonce technique as the users query to avoid the problem of racing results showing the wrong result.

Reply:
I tried the nonce technique and it was just super slow for some reason